Upstream-Status: Pending
diff --git a/dts/upstream/src/arm64/ti/k3-am625-sk.dts b/dts/upstream/src/arm64/ti/k3-am625-sk.dts
index b1161e688bf..60b7ac33b82 100644
--- a/dts/upstream/src/arm64/ti/k3-am625-sk.dts
+++ b/dts/upstream/src/arm64/ti/k3-am625-sk.dts
@@ -14,7 +14,7 @@
 
 / {
 	compatible = "ti,am625-sk", "ti,am625";
-	model = "Texas Instruments AM625 SK";
+	model = "@UBOOT_BOOT_DEVICE@ AM625";
 
 	opp-table {
 		/* Add 1.4GHz OPP for am625-sk board. Requires VDD_CORE to be at 0.85V */
