diff --git a/arch/arm/dts/k3-am625-sk-binman.dtsi b/arch/arm/dts/k3-am625-sk-binman.dtsi
index 3eab1522664..a54ffba032c 100644
--- a/arch/arm/dts/k3-am625-sk-binman.dtsi
+++ b/arch/arm/dts/k3-am625-sk-binman.dtsi
@@ -55,7 +55,6 @@
 &binman {
 	tiboot3-am62x-hs-fs-evm.bin {
 		filename = "tiboot3-am62x-hs-fs-evm.bin";
-		symlink = "tiboot3.bin";
 		ti-secure-rom {
 			content = <&u_boot_spl_fs>, <&ti_fs_enc_fs>, <&combined_tifs_cfg_fs>,
 				<&combined_dm_cfg_fs>, <&sysfw_inner_cert_fs>;
@@ -101,6 +100,7 @@
 &binman {
 	tiboot3-am62x-gp-evm.bin {
 		filename = "tiboot3-am62x-gp-evm.bin";
+		symlink = "tiboot3.bin";
 		ti-secure-rom {
 			content = <&u_boot_spl_unsigned>, <&ti_fs_gp>,
 				<&combined_tifs_cfg_gp>, <&combined_dm_cfg_gp>;
diff --git a/arch/arm/dts/k3-am625-sk.dts b/arch/arm/dts/k3-am625-sk.dts
index 823e4320c3c..b7eb22ca522 100644
--- a/arch/arm/dts/k3-am625-sk.dts
+++ b/arch/arm/dts/k3-am625-sk.dts
@@ -14,7 +14,7 @@
 
 / {
 	compatible = "ti,am625-sk", "ti,am625";
-	model = "Texas Instruments AM625 SK";
+	model = "WARP AM625";
 
 	aliases {
 		serial2 = &main_uart0;
@@ -141,7 +141,6 @@
 		regulator-boot-on;
 		enable-active-high;
 		vin-supply = <&vcc_3v3_sys>;
-		gpio = <&exp1 3 GPIO_ACTIVE_HIGH>;
 	};
 
 	vdd_sd_dv: regulator-4 {
diff --git a/board/ti/am62x/am62x.env b/board/ti/am62x/am62x.env
index e507ca276ce..7c944ee63ac 100644
--- a/board/ti/am62x/am62x.env
+++ b/board/ti/am62x/am62x.env
@@ -7,20 +7,10 @@
 #include <environment/ti/k3_rproc.env>
 #endif
 
-default_device_tree=ti/k3-am625-sk.dtb
+default_device_tree=warp/warp-625x.dtb
 findfdt=
 	setenv name_fdt ${default_device_tree};
-	if test $board_name = am62x_skevm; then
-			setenv name_fdt ti/k3-am625-sk.dtb; fi;
-	if test $board_name = am62b_p1_skevm; then
-			setenv name_fdt ti/k3-am625-sk.dtb; fi;
-	if test $board_name = am62x_sip_skevm; then
-                        setenv name_fdt ti/k3-am625-sk.dtb; fi;
-	if test $board_name = am62x_lp_skevm; then
-			setenv name_fdt ti/k3-am62-lp-sk.dtb; fi;
-	if test $board_name = am62x_beagleplay; then
-			setenv name_fdt ti/k3-am625-beagleplay.dtb; fi;
-	setenv fdtfile ${name_fdt}
+
 name_kern=Image
 console=ttyS2,115200n8
 args_all=setenv optargs ${optargs} earlycon=ns16550a,mmio32,0x02800000
